File: adonis/app/Http/Controllers/TodosController.js
'use strict'

const Todo = use('App/Model/Todo')

class TodosController {

  * index (request, response) {
    const todos = yield Todo.all()
    response.json(todos)
  }

  * show (request, response) {
    const todo = yield Todo.find(request.param('id'))
    response.json(todo)
  }

  * store (request, response) {
    const todo = yield Todo.create({
      name: request.input('name')
    })
    response.json(todo)
  }

  * update (request, response) {
    const todo = yield Todo.find(request.param('id'))
    yield todo.fill({
      name: request.input('name')
    })
    response.json(todo)
  }

  * toggle (request, response) {
    const todo = yield Todo.find(request.param('id'))
    todo.completed = !todo.completed
    yield todo.save()
    response.json(todo)
  }

}

module.exports = TodosController

File: ruby/app/controllers/todos_controller.rb
class TodosController < ApplicationController
  def index
    todos = Todo.all

    render json: todos, status: :ok
  end

  def show
    todo = Todo.find(params[:id])

    render json: todo, status: :ok
  end

  def create
    todo = Todo.create!(person_params)

    render json: todo, status: :ok
  end

  def update
    todo = Todo.find(params[:id])
    todo.update!(person_params)

    render json: todo, status: :ok
  end

  def destroy
    todo = Todo.find(params[:id])
    todo.destroy

    head :no_content
  end

  private
    def person_params
      params.permit(:name)
    end
end
